documentation/modules/ROOT/partials/modules/mysql-connector/con-mysql-connector-events.adoc

```diff
@@ -428,28 +548,50 @@ The value of a _delete_ change event on the `customers` table has the exact same
+If the {link-prefix}:{link-mysql-connector}#enable-query-log-events-for-cdc_{context}[`binlog_rows_query_log_events`] MySQL configuration option is enabled and the connector configuration `include.query` property is enabled, the `source` field also provides the `query` field, which contains the original SQL statement that caused the change event.
 
-When a row is deleted, the _delete_ event value listed above still works with log compaction, because Kafka can still remove all earlier messages with that same key. If the message value is `null`, Kafka knows that it can remove all messages with that same key. To make this possible, {prodname}’s MySQL connector always follows a _delete_ event with a special tombstone event that has the same key but a `null` value.
+|4
+|`op`
+a|Mandatory string that describes the type of operation. The `op` field value is `d`, signifying that this row was deleted.
 
-== Primary Key Update Header
+|5
+|`ts_ms`
+a|Optional field that displays the time at which the connector processed the event. The time is based on the system clock in the JVM running the Kafka Connect task.
+
+|===
 
-When there is an update event that's changing the row's primary key field/s, also known
-as a primary key change, Debezium will in that case send a DELETE event for the old key
-and an INSERT event for the new (updated) key.
+A _delete_ change event record provides a consumer with the information it needs to process the removal of this row. The old values are included because some consumers might require them in order to properly handle the removal.
 
-The DELETE event produces a Kafka message which has a message header `__debezium.newkey`
-and the value is the new primary key.
+MySQL connector events are designed to work with link:https://kafka.apache.org/documentation/#compaction[Kafka log compaction]. Log compaction enables removal of some older messages as long as at least the most recent message for every key is kept. This lets Kafka reclaim storage space while ensuring that the topic contains a complete data set and can be used for reloading key-based state.
 
-The INSERT event produces a Kafka message which has a message header `__debezium.oldkey`
-and the value is the previous (old) primary key of the updated row.
+[id="mysql-tombstone-events"]
+.Tombstone events
+When a row is deleted, the _delete_ event value still works with log compaction, because Kafka can remove all earlier messages that have that same key. However, for Kafka to remove all messages that have that same key, the message value must be `null`. To make this possible, {prodname}’s MySQL connector follows a _delete_ event with a special tombstone event that has the same key but a `null` value.
```

Comment:
Interesting, never saw that usage of "follow" in the last sentence. I knew about "B follows A" and "from A follows B", but not about "someone follows something with another thing". TIL :)

Reply:
It is correct - though I did have to think about it! So I decided to make it easier/quicker to understand and changed it to this: 

To make this possible, after {prodname}’s MySQL connector emits a _delete_ event, the connector emits a special tombstone event that has the same key but a `null` value.